plugin/completion.py

```diff
@@ -72,274 +111,81 @@ def initialize(self) -> None:
-            if additional_edits:
-                self.apply_additional_edits(additional_edits)
-            elif self.resolve:
-                self.do_resolve(item)
-
-        else:
-            debug('could not find completion item for inserted "{}"'.format(inserted))
+        self.manager.documents.purge_changes(self.view)
+        document_position = text_document_position_params(self.view, locations[0])
+        client.send_request(
+            Request.complete(document_position),
+            lambda res: self.handle_response(res, completion_list, prefix),
+            self.handle_error)
 
-    def match_selector(self, location: int) -> bool:
-        return self.view.match_selector(location, self.auto_complete_selector)
-
-    def on_query_completions(self, prefix: str, locations: List[int]) -> Optional[Tuple[List[Tuple[str, str]], int]]:
-        if not self.initialized:
-            self.initialize()
+    def handle_response(self, response: Optional[Union[dict, List]],
+                        completion_list: sublime.CompletionList, prefix: str) -> None:
+        response_items, response_incomplete = parse_completion_response(response)
+        items = list(format_completion(item) for item in response_items)
 
         flags = 0
         if settings.only_show_lsp_completions:
             flags |= sublime.INHIBIT_WORD_COMPLETIONS
             flags |= sublime.INHIBIT_EXPLICIT_COMPLETIONS
 
-        if self.enabled:
-            if not self.match_selector(locations[0]):
-                return ([], flags)
-
-            reuse_completion = self.is_same_completion(prefix, locations)
-            if self.state == CompletionState.IDLE:
-                if not reuse_completion:
-                    self.last_prefix = prefix
-                    self.last_location = locations[0]
-                    self.do_request(prefix, locations)
-                    self.completions = []
-
-            elif self.state in (CompletionState.REQUESTING, CompletionState.CANCELLING):
-                if not reuse_completion:
-                    self.next_request = (prefix, locations)
-                    self.state = CompletionState.CANCELLING
-
-            elif self.state == CompletionState.APPLYING:
-                self.state = CompletionState.IDLE
-
-            return (self.completions, flags)
-
-        return None
-
-    def on_text_command(self, command_name: str, args: Optional[Any]) -> None:
-        self.committing = command_name in ('commit_completion', 'auto_complete')
-
-    def do_request(self, prefix: str, locations: List[int]) -> None:
-        self.next_request = None
-        view = self.view
-
-        # don't store client so we can handle restarts
-        client = client_from_session(session_for_view(view, 'completionProvider', locations[0]))
-        if not client:
-            return
-
-        if settings.complete_all_chars or self.is_after_trigger_character(locations[0]):
```

Comment:
Where did this functionality go? Are we always requesting completions now?

Reply:
Not always, only when on_query_completion is called. (I noted that are all AZ-az chars, and all `auto_complete_triggers` chars)